pkg/apis/v1beta1/cluster.go

```diff
@@ -127,28 +127,35 @@ func validateSpecs(v Validateable) []error {
 	return v.Validate()
 }
 
-// FromYamlFile ...
-func FromYamlFile(filename string, k0sVars constant.CfgVars) (*ClusterConfig, error) {
+// ConfigFromFile ...
+func ConfigFromFile(filename string, k0sVars constant.CfgVars) (*ClusterConfig, error) {
 	buf, err := ioutil.ReadFile(filename)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read config file at %s: %w", filename, err)
 	}
 
-	return FromYamlString(string(buf), k0sVars)
+	return configFromString(string(buf), k0sVars)
 }
 
-// FromYamlPipe
-func FromYamlPipe(r io.Reader, k0sVars constant.CfgVars) (*ClusterConfig, error) {
-	input, err := ioutil.ReadAll(r)
+func isInputFromPipe() bool {
+	fi, _ := os.Stdin.Stat()
+	return fi.Mode()&os.ModeCharDevice == 0
+}
+
+// ConfigFromStdin tries to read k0s.yaml config from stdin
+func ConfigFromStdin(k0sVars constant.CfgVars) (*ClusterConfig, error) {
+	if !isInputFromPipe() {
```

Comment:
Do we really need this test?

What happens if you do: `k0s ... --config - <./path/to/file`

Reply:
> k0s ... --config - <./path/to/file

this will read the config from file.

> Do we really need this test?
actually it would either way just fall on the next line, so not that much difference if we drop it (and fix return nil, nil on the next line, which is a bug).